test/torch_basics_test.py

```diff
@@ -53,6 +54,7 @@ class BasicsTests(unittest.TestCase):
     def __init__(self, *args, **kwargs):
         super(BasicsTests, self).__init__(*args, **kwargs)
         warnings.simplefilter("module")
+        os.environ['BLUEFOG_NODES_PER_MACHINE'] = '2'
```

Comment:
Is it possible to move this to the two individual test cases below, as we probably don't want to make all the tests here hierarchical?

Reply:
Good point. But there is a difficulty here. Local_comm and cross_comm are built during the init() call and we can only call the init() once. If we don't put it in the top-level, it may never be able to control it again. I can leave it as your feature request to implement it, haha.

Reply:
Discussed offline. Will move every hierarchical related test into another file.